### DISCUSSION TITLE
Image Component Optimization not working without fixed height

### DISCUSSION THREAD

**Discussion by MejanH** [vercel/next.js#19027] [Help]
First of all, Thanks for the new Nextjs update with amazing features. Never regretted any Nextjs feature till now.

I have tried with `layout="fill"` option. No matter what I do, I still have to add a fixed height to make it work. I am good with the width option. but height with `object-fit:...` style cropping my image or leaving extra space with `object-fit: contain;`.

I want to display the full image where my images are in different sizes.

```Javascript
      <div className=" relative">
        <Image
          layout="fill"
          src={`/images/${cover}`}
          className=" rounded-xl mx-auto"
        />
      </div>
```

**Comment by irg1008** [comment]
Same problem here. As far as I know you need to set height and width or stick with the cover/contain options. Next Image cannot calculate the remain attributte of the image.
Problem with layout fill is that you lose Image optimization for some reason.
